services/QuillLMS/app/models/subscription.rb

```diff
@@ -165,13 +169,18 @@ def credit_user_and_de_activate
       report_to_new_relic("Sub credited and expired with multiple users. Subscription: #{id}")
     else
       update(de_activated_date: Date.current, recurring: false)
+      stripe_cancel_at_period_end
       # subtract later of start date or today's date from expiration date to calculate amount to credit
       # amount_to_credit = self.expiration - [self.start_date, Date.current].max
       amount_to_credit = expiration - start_date
       CreditTransaction.create(user_id: user_subscriptions.first.user_id, amount: amount_to_credit.to_i, source: self)
     end
   end
 
+  def disable_stripe_auto_renew
+    Stripe::Subscription.update(stripe_subscription_id, cancel_at_period_end: true)
+  end
```

Comment:
Given that you're doing the actually `update` through a delegation, is this stand-alone method still required on the class?  I don't see it being called anywhere.

Reply:
Good catch.  I completely missed this.